handlers/accesscontrol/access-control.go

```diff
@@ -124,18 +124,23 @@ func (ac *AccessControlHandlersCollection) periodicRefreshIntervalCache(mapic mi
 			time.Sleep(5 * time.Second)
 			ac.mutex.Lock()
 			refreshIntervalCache.mux.Lock()
+			var keysToInvalidate []string
 			for key := range refreshIntervalCache.data {
 				if time.Since(refreshIntervalCache.data[key].LastRefresh) > time.Duration(refreshIntervalCache.data[key].RefreshInterval)*time.Second {
 					refreshIntervalCache.data[key].LastRefresh = time.Now()
-					mapic.InvalidateAllSessions(key)
+					keysToInvalidate = append(keysToInvalidate, key)
 					for cachedAccessKey := range ac.cache[key] {
 						delete(ac.cache[key], cachedAccessKey)
 					}
 					break
 				}
 			}
-			ac.mutex.Unlock()
 			refreshIntervalCache.mux.Unlock()
+			ac.mutex.Unlock()
+
+			for _, key := range keysToInvalidate {
+				mapic.InvalidateAllSessions(key)
```

Comment:
plural "sessions" suggests that it might allow multiple sessions being passed at once... https://github.com/livepeer/mistserver/blob/catalyst/src/controller/controller_api.cpp#L1127-L1129
but I know it's not the scope of your focus atm

Reply:
Yeah, I don't want to touch it, because I want to focus on one thing (which is fixing this weird deployment issue).

Reply:
I think it means invalidate all playback sessions for that stream, so only able to process one streamID at a time